bindings/java/src/main/native/endpoint.cc

```diff
@@ -204,6 +204,31 @@ Java_org_openucx_jucx_ucp_UcpEndpoint_sendTaggedNonBlockingNative(JNIEnv *env, j
     return process_request(request, callback);
 }
 
+JNIEXPORT jobject JNICALL
+Java_org_openucx_jucx_ucp_UcpEndpoint_sendTaggedIovNonBlockingNative(JNIEnv *env, jclass cls,
+                                                                    jlong ep_ptr, jlongArray addresses,
+                                                                    jlongArray sizes, jlong tag,
+                                                                    jobject callback)
+{
+    jsize iovcnt = env->GetArrayLength(addresses);
```

Comment:
check that addresses and sizes have same length

Reply:
Done in java